File: Definitions/Game_Definitions/classes/DamageTypeHitPrereqCondition.lua
---@meta
---@diagnostic disable

---@class DamageTypeHitPrereqCondition : BaseHitPrereqCondition
---@field damageType gamedataDamageType
DamageTypeHitPrereqCondition = {}

---@return DamageTypeHitPrereqCondition
function DamageTypeHitPrereqCondition.new() return end

---@param props table
---@return DamageTypeHitPrereqCondition
function DamageTypeHitPrereqCondition.new(props) return end

---@param hitEvent gameeventsHitEvent
---@return Bool
function DamageTypeHitPrereqCondition:Evaluate(hitEvent) return end

---@param recordID TweakDBID|string
function DamageTypeHitPrereqCondition:SetData(recordID) return end

File: DecompiledGameScripts/core/gameplay/prereqs/damage/damageTypeHitPrereqCondition.swift

public class DamageTypeHitPrereqCondition extends BaseHitPrereqCondition {

  public let m_damageType: gamedataDamageType;

  public func SetData(recordID: TweakDBID) -> Void {
    let str: String = TweakDBInterface.GetString(recordID + t".damageType", "");
    this.m_damageType = IntEnum<gamedataDamageType>(Cast<Int32>(EnumValueFromString("gamedataDamageType", str)));
    super.SetData(recordID);
  }

  public func Evaluate(hitEvent: ref<gameHitEvent>) -> Bool {
    let result: Bool = hitEvent.attackComputed.GetAttackValue(this.m_damageType) > 0.00;
    if result {
      result = this.CheckOnlyOncePerShot(hitEvent);
    };
    return this.m_invert ? !result : result;
  }
}
